### DISCUSSION TITLE
Forges Addon and Blacksmith Upgrade?

### DISCUSSION THREAD

**Discussion by OcelotWalrus** [Dungeons-of-Kathallion/Bane-Of-Wargs#14] [Ideas]
I feel like blacksmith need to do more stuff, but I can't find anything.

For completing blacksmiths, there should be forges, that sells/buys you metals. But maybe it should be more that that?

**Comment by lumbar527** [comment]
Maybe (though this would add a lot more complication or extra items) blacksmiths/forges could upgrade some of your weapons for money.

**Reply by OcelotWalrus** [reply]
I already had that idea but I don't really know if it should be added. It's easy to do but it would make a drastic change to the game. 

But know that I'm thinking of it, making weapons unique and upgradable could be interesting and add more fun to the game.

**Reply by lumbar527** [reply]
I have an idea of how to do that—can I make a pull request for it?

**Reply by lumbar527** [reply]
I'm working on making an inventory file to go with your save that would make items more unique.

**Reply by OcelotWalrus** [reply]
Sure!

**Comment by OcelotWalrus** [comment]
> I'm working on making an inventory file to go with your save that would make items more unique.

Actually I would prefer that instead of creating a new file just for the inventory, you just make so that the `inventory` key is now a dict instead of a list.

**Reply by OcelotWalrus** [reply]
If you want I can take care of the engine conversion stuff. I made a branch for the development: [item-remaster](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/tree/item-remaster). So you don't need to make a PR for that and we can collaborate easily.

**Reply by lumbar527** [reply]
I already made a branch and started working, and I think it makes more sense to manage it in a different file. Also, we shouldn't be directly committing to the same branch.

**Reply by OcelotWalrus** [reply]
There is no reason for making a file just for the inventory. I would really prefer to keep it in the save file.

**Reply by lumbar527** [reply]
OK, going to start working on `item-remaster` now.

**Reply by OcelotWalrus** [reply]
Thanks! I really appreciate that!

**Comment by OcelotWalrus** [comment]
I've got an idea: Maybe instead of completely modifying most of the engine for a new item system, we could make so that weapons have multiple definitions that defines how many time you can upgrade the weapon:
```
# create a key template that will be the basic weapon item
Broadsword: &broadsword
  display name: "Broadsword"
  type: "Weapon"
  damage: 7
  defend: 6
  agility: .04
  gold: 12.98
  critical hit chance: .09
  description "The Broadsword is a large sword and powerful sword that not anyone can use well."
Broadsword (1):
  << : *broadsword
  damage: 8
  defend: 7
  agility: .045
  critical hit chance: .095
  for this upgrade:
  - Raw Iron
  - Raw Iron
  - Raw Iron
  - Raw Copper
Broadsword (2):
  << : *broadsword
  damage: 9.3
  defend: 7.6
  agility: 0.48
  critical hit chance: .1
  for this upgrade:
  - Raw Iron
  - Raw Iron
  - Raw Copper
  - Raw Silver
```
Because I feel like the first idea will make the engine change too much. And it is too complicated for something pretty simple.

**Reply by OcelotWalrus** [reply]
So upgrade an weapon in a blacksmith/forge will just update the weapon.

**Reply by lumbar527** [reply]
Sure! That was one of the things I was thinking of; either multiple weapons or being able to change weapon data.

**Reply by OcelotWalrus** [reply]
So it's decided, weapons will work like that. I think I'd prefer to code everything myself.

**Reply by lumbar527** [reply]
That's fine. You can either work on the new branch (and reset it so my commit isn't there) or just do it wherever you want.

**Comment by lumbar527** [comment]
Is this resolved or do you want to discuss it more?

**Comment by OcelotWalrus** [comment]
I think it's resolved.

**Reply by OcelotWalrus** [reply]
I'm going to work on churches then the map item.
